MultipleValidatorWithAnd now can break out of loop when error occurs

issei-m opened the pull request:
I often use `MultipleValidatorWithAnd` to combine validations such as `RFCValidation`, `DNSCheckValidation`.
In this case, when the preceding RFC check fails, I wanna skip the succeeding DNS check process since that would be costly.

That's why I would like to add a single option for this class to break out of validation loop when one of validations gets failure.

egulias commented:
How about adding a constant as a flag so the API has more meaning. E.g
 ```php
const STOP_ON_ERROR = true;
const ALLOW_ALL_ERRORS = false;

public function __construc(array $validatons, $breakCondition = ALLOW_ALL_ERRORS)
```

egulias commented on EmailValidator/Validation/MultipleValidationWithAnd.php:
```diff
@@ -3,23 +3,49 @@
+    /**
+     * @param EmailValidation[] $validations  The validations.
+     * @param bool              $breakIfError If true, it breaks out of validation loop when error occurs,
+     *                                        it means returned MultipleErrors might not contain all causes of errors. (false by default)
```
`it means returned MultipleErrors will contain a single error, the first found.`

egulias commented on EmailValidator/Validation/MultipleValidationWithAnd.php:
```diff
@@ -29,17 +55,27 @@ public function isValid($email, EmailLexer $emailLexer)
             $this->warnings = array_merge($this->warnings, $validation->getWarnings());
             $errors[] = $validation->getError();
+
+            if (!$result && $this->breakIfError) {
```
WDYT of encapsulating into a method? Like
```php
if($this->shouldStop($result)) {
    break;
}
```

egulias commented on EmailValidator/Validation/MultipleValidationWithAnd.php:
```diff
@@ -3,23 +3,49 @@
 class MultipleValidationWithAnd implements EmailValidation
 {
+    /**
+     * @var EmailValidation[]
```
Not sure here. Shouldn't this be
`@var array EmailValidation`?

issei-m replied:
Intends to make IDE's analyzer smarter, each elements of `$validations` now can be guessed its type in the loop block like this:

![image](https://cloud.githubusercontent.com/assets/1135118/16621299/b55a440a-43d0-11e6-85db-be83ebd7cfd1.png)

issei-m commented:
@egulias Updated now to change following:

- renamed `$breakIfError` option to `$mode`
- put constants for $mode
- extracted skip logic into private method

egulias commented:
I'll issue a new release next weekend, thanks!
